src/init.cpp

```diff
@@ -1015,6 +1015,13 @@ bool AppInitParameterInteraction(const ArgsManager& args)
         }
     }
 
+    if (args.IsArgSet("-blockmaxweight")) {
+        const auto max_block_weight = args.GetIntArg("-blockmaxweight", DEFAULT_BLOCK_MAX_WEIGHT);
+        if (max_block_weight > DEFAULT_BLOCK_MAX_WEIGHT) {
+            return InitError(strprintf(_("Specified -blockmaxweight (%d) exceeds policy default maximum block weight (%d)"), max_block_weight, DEFAULT_BLOCK_MAX_WEIGHT));
```

Comment:
In 8b8e143a8fdd9c12068c7d7f5bbe5317dd55d22a "init: fail to start when `-blockmaxweight` exceeds `DEFAULT_MAX_BLOCK_WEIGHT`": I think this should only check against consensus.

Reply:
In block assembly, we check against policy. If we were to lower the policy default, users could set a higher value than our policy limit, but the generated block would still be checked against policy, meaning their higher value would have no effect. I think we should be consistent and either check against policy everywhere or remove the policy default and check only against consensus?

Reply:
Policy is what we put in blocks by default (`DEFAULT_BLOCK_MAX_WEIGHT`). If the user wants to override the default using `-blockmaxweight`, they should only be limited by consensus (`MAX_BLOCK_WEIGHT`).

> but the generated block would still be checked against policy, meaning their higher value would have no effect.

That would be a bug.

Reply:
Fixed!